Array.length treated as `@UnknownInitialization`

Consider the following test case:

```java
import org.checkerframework.framework.qual.AnnotatedFor;

@AnnotatedFor("nullness")
public class Issue3449 {

  int length;

  public Issue3449(Object... args) {
    length = args.length;
  }
}
```

Compiling it with

```
javac -g Issue3449.java -processor nullness -AuseConservativeDefaultsForUncheckedCode=source
```

yields this error:

```
Issue3449.java:9: error: [assignment.type.incompatible] incompatible types in assignment.
    length = args.length;
                 ^
  found   : @UnknownInitialization @NonNull int
  required: @Initialized @NonNull int
1 error
```

The file type-checks if I remove `@AnnotatedFor` or the `-AuseConservativeDefaultsForUncheckedCode=source` command-line option.

--- a/checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java
+++ b/checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java
@@ -3,6 +3,7 @@ package org.checkerframework.checker.initialization;
 import com.sun.source.tree.ClassTree;
 import com.sun.source.tree.ExpressionTree;
 import com.sun.source.tree.LiteralTree;
+import com.sun.source.tree.MemberSelectTree;
 import com.sun.source.tree.MethodTree;
 import com.sun.source.tree.NewClassTree;
 import com.sun.source.tree.Tree;
@@ -741,6 +742,15 @@ public abstract class InitializationAnnotatedTypeFactory<
                 type.addAnnotation(INITIALIZED);
             }
             return super.visitLiteral(tree, type);
+        }
+
+        @Override
+        public Void visitMemberSelect(
+                MemberSelectTree node, AnnotatedTypeMirror annotatedTypeMirror) {
+            if (TreeUtils.isArrayLengthAccess(node)) {
+                annotatedTypeMirror.replaceAnnotation(INITIALIZED);
+            }
+            return super.visitMemberSelect(node, annotatedTypeMirror);
         }
     }
 

--- a/checker/tests/nullness-safedefaultssourcecode/Issue3449.java
+++ b/checker/tests/nullness-safedefaultssourcecode/Issue3449.java
@@ -1,7 +1,5 @@
 // Test case for https://tinyurl.com/cfissue/3449
 
-// @skip-test until the issue is fixed
-
 import org.checkerframework.framework.qual.AnnotatedFor;
 
 @AnnotatedFor("nullness")

--- a/checker/tests/nullness-safedefaultssourcecode/Issue3449.java
+++ b/checker/tests/nullness-safedefaultssourcecode/Issue3449.java
@@ -8,8 +8,10 @@ import org.checkerframework.framework.qual.AnnotatedFor;
 public class Issue3449 {
 
     int length;
+    Object[] objs;
 
     public Issue3449(Object... args) {
         length = args.length;
+        objs = args;
     }
 }

